run/README.md

```diff
@@ -0,0 +1,131 @@
+
+[Cloud Run][run_docs] stateless containers on a fully managed environment or in your own GKE cluster.
+
+## Samples
+
+|           Sample                |        Description       |     Deploy    |
+| ------------------------------- | ------------------------ | ------------- |
+|[Hello World][helloworld]&nbsp;&#10149; | Quickstart | [<img src="https://storage.googleapis.com/cloudrun/button.svg" alt="Run on Google Cloud" height="30">][run_button_helloworld] |
+|[Manual Logging][manual_logging] | Structured loggin without client library | [<img src="https://storage.googleapis.com/cloudrun/button.svg" alt="Run on Google Cloud" height="30">][run_button_manual_logging] |
+
+For more Cloud Run samples beyond Nodejs, see the main list in the [Cloud Run Samples repository](https://github.com/GoogleCloudPlatform/cloud-run-samples).
+
+## Setup
+
+1. [Set up for Cloud Run development](https://cloud.google.com/run/docs/setup)
+
+2. Clone this repository:
+
+    ```
+    git clone https://github.com/GoogleCloudPlatform/nodejs-docs-samples.git
+    ```
+
+    Note: Some samples in the list above are hosted in other repositories. They are noted with the symbol "&#10149;".
+
+
+## How to run a sample locally
+
+1. [Install docker locally](https://docs.docker.com/install/)
+
+2. [Build the sample container](https://cloud.google.com/run/docs/building/containers#building_locally_and_pushing_using_docker):
+
+    ```
+    export SAMPLE=$sample
+    cd $SAMPLE
+    docker build --tag $sample .
+    ```
+
+3. [Run containers locally](https://cloud.google.com/run/docs/testing/local)
+
+    With the built container:
+
+    ```
+    PORT=8080 && docker run --rm -p 8080:${PORT} -e PORT=${PORT} $SAMPLE
+    ```
+
+    Overriding the built container with local code:
+
+    ```
+    PORT=8080 && docker run --rm \
+        -p 8080:${PORT} -e PORT=${PORT} \
+        -v $PWD:/usr/src/app $SAMPLE
+    ```
+
+    Injecting your service account key:
+
+    ```
+    export SA_KEY_NAME=my-key-name-123
+    PORT=8080 && docker run --rm \
+        -p 8080:${PORT} -e PORT=${PORT} \
+        -e GOOGLE_APPLICATION_CREDENTIALS=/tmp/keys/${SA_KEY_NAME}.json \
+        -v $GOOGLE_APPLICATION_CREDENTIALS:/tmp/keys/${SA_KEY_NAME}.json:ro \
+        -v $PWD:/usr/src/app $SAMPLE
+    ```
+
+    Opening a shell in the container (e.g., updating the `package-lock.json`):
+
+    1. Build the container.
+
+    2. Run the container with a shell:
+
+    ```
+    PORT=8080 && docker run --rm \
+        --interactive --tty \
+        -p 8080:${PORT} -e PORT=${PORT} \
+        -v $PWD:/usr/src/app $SAMPLE \
+        /bin/bash
+    ```
+    3. Re-generate the `package-lock.json`:
+
+    ```
+    rm package-lock.json
+    npm install
+    ```
+
+    Because we're using a read/write volume mount, the revised file will be
+    written to the host's local filesystem. Once you exit the container you can
+    add the file to version control.
+
+    4. Exit the container: ^D
```

Comment:
Nit: use `Ctrl-D` here instead of `^D`? (Not everyone may realize that `^` == `Ctrl`)